spyder/plugins/ipythonconsole/widgets/debugging.py

```diff
@@ -196,3 +217,21 @@ def _event_filter_console_keypress(self, event):
     def is_debugging(self):
         """Check if we are debugging."""
         return self.spyder_kernel_comm._debugging
+
+    def add_to_pdb_history(self, line_num, line):
+        """Add command to history"""
+        self._control.histidx = None
```

Comment:
The same for this: Why not make `histidx` an attribute of `DebuggingWidget`?

Reply:
Same reason, BrowseHistory is a parent of the control.